department-of-veterans-affairs/caseflow
# APPEALS/30242 - Create `ReportPage` for Generate Task Report Epic

Resolves https://jira.devops.va.gov/browse/APPEALS-30242
# Description
1. Refactor NonComp pages for more design flexibility
The NonComp pages have been refactored to allow for buttons outside of the `AppSegment` component.
Throughout the app we are moving to this design pattern and want the flexibility to allow for that without any hacky css in these pages.

2. Update the `decision_reviews` routing
After discussion we decided it was better to do the exact routing through React Router on the frontend.
Instead of adding in a new exact path for each new decision_review page we create we can point all decision_review pages to the index view which will display the main DecisionReview component on the frontend.
From there we can get more granular and break up exact paths into separate components.

3.  Create the basic structure for ReportPage that will be expanded on as more stories are completed.


## Acceptance Criteria
- [ ] Only VHA admins can access page.
- [ ] User can return to VHA Decision Review Queue by clicking ‘Cancel’ link.
- [ ] Header to display 'Generate task report'. 
- [ ] Non-functional buttons are created for 'Clear filters' and 'Generate task report'
- [ ] The page at `/decision_reviews/vha/tasks/{taskId}` displays correctly and has unchanged functionality
- [ ] The page at `/decision_reviews/` displays correctly and has unchanged functionality

## Testing Plan
Report Page:
You can access the page by manually navigating to `/decision_reviews/vha/report`
- Test the cancel button navigates you back correctly

Task Page:
At `/decision_reviews/vha/tasks/{taskId}`
- Make sure the formatting all looks the same as before
- Make sure the functionality is the same as best we can

Decision Review Page:
At `/decision_reviews`
- Make sure the formatting all looks the same as before
- Make sure the functionality is the same as best we can
- Test that the banner displays correctly at the top same as before. The easiest way to test this is by creating/deleting/saving an issue on the claim and submitting. This will redirect you back to the `/decision_reviews` url where you should see a banner at the top of the page.


## Screenshot
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/51007432/2b685240-512f-4944-8418-872bfb9e350f)
